src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -121,38 +128,46 @@ public PackageObjectFactory(String packageName, ClassLoader moduleClassLoader) {
      */
     @Override
     public Object createModule(String name) throws CheckstyleException {
-        Object instance = createObjectFromMap(name);
+        final String nameCheck = name + CHECK_SUFFIX;
+        Object instance = createObject(name);
```

Comment:
> They are only looking up names in the map and it should never cleared.

I see now that the list isn't static. I was thinking it was but it can't be static since it is based on the values being sent to the constructor. I will make new item since this is hidden now.